### PR TITLE
KIALI-1218 Service details: Pods and deployments removed + adding workloads

### PR DESCRIPTION
**Pull request by xeviknal**
** Describe the change **

Adapted the modifications performed into ServiceDetails endpoint: https://github.com/kiali/kiali/pull/418
List of pods and deployments are removed. Workloads added.

Also there is a proposal on moving the Istio icon next to service title (as suggested in the comments)

This PR needs to be deployed at the same time as https://github.com/kiali/kiali/pull/418

** Issue reference **

[KIALI-1304](https://issues.jboss.org/browse/KIALI-1304): Update ServiceDetails endpoint.

** Backwards in compatible? **
Yes.

** Screenshot **

If Istio is enabled for the service:
screenshot of kiali console 9

Otherwise:
screenshot of kiali console 10

### CODE REVIEW

**Comment by abonas** [comment]
I think that "Istio sidecard" indication should be removed (sidecar is not deployed on service), now this is even more confusing.

**Comment by xeviknal** [comment]
I'll keep on working this PR in order to have workload list as an extra tab.

**Comment by lucasponce** [comment]
@xeviknal +1 with your comment
Services should validate now the "app" flag to indicate that are part of "Istio" ecosystem (wording should be refined), but yes, as a user I want to filter those Services that are part of Istio and warn when the "app" label is not present.

I agree that "Istio Deployed" concept belongs to the Workload too, but we want to maintain that differentiation on the services area as well.

To be defined if using the same Istio logo or a different one is the good option as well, but until UX decide, conceptually we want to mark the service with that distinction (and let user filters them in the previous service list page).

**Comment by abonas** [comment]
> @xeviknal +1 with your comment
> Services should validate now the "app" flag to indicate that are part of "Istio" ecosystem (wording should be refined), but yes, as a user I want to filter those Services that are part of Istio and warn when the "app" label is not present.

I think relying on "app" label can be not very accurate. noone said that only Istio users use this label. You could very likely have users putting an app label without using Istio.

> 
> I agree that "Istio Deployed" concept belongs to the Workload too, but we want to maintain that differentiation on the services area as well.

I like the proposals to have an Istio indicator, without the "deployed" concept

> 
> To be defined if using the same Istio logo or a different one is the good option as well, but until UX decide, conceptually we want to mark the service with that distinction (and let user filters them in the previous service list page).

**Comment by xeviknal** [comment]
@abonas @lucasponce @jotak Right now, this is just a proposal of the changes.
1. How to display de istio icon to remove misleadings.
2. How to display Workloads (my other alternative was to use the tables we have for Virtual Services and Destination Rules).

Ill appreciate your suggestions :)

File: src/pages/ServiceDetails/ServiceInfo/ServiceInfoWorkload.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    return (
+      <div className="card-pf">
+        <Row className="row-cards-pf">
+          <Col xs={12} sm={12} md={12} lg={12}>
+            {(this.props.workloads || []).map((workload, u) => (
+              <div className="card-pf-body" key={'workloads_' + u}>
+                <h3>{workload.name}</h3>
+                <div key="labels" className="label-collection">
+                  {Object.keys(workload.labels || {}).map((key, i) => (
+                    <Label key={'workload_' + i} name={key} value={workload.labels ? workload.labels[key] : ''} />
```
Comment (jotak) [inline src/pages/ServiceDetails/ServiceInfo/ServiceInfoWorkload.tsx]: Maybe that should be a link to the relative workload details page?

**Comment by xeviknal** [comment]
Great!
About the link to workload details page, I preferred not to add it because I would have to set the href to an invalid location, and IMO, it would be a bug. I add a comment to [KIALI-1217](https://issues.jboss.org/browse/KIALI-1217) (Creation of the workload details page) in order to remind this addition.

**Comment by xeviknal** [comment]
In order to be more consistent with Virtual Services and Destination Rules lists, I moved the information into a table with same look and feel.

screenshot of kiali console 11

There is still missing the link to workload details page.

**Comment by lucasponce** [comment]
This PR has some conflicts and it needs to be added together with https://github.com/kiali/kiali/pull/418.
So, @xeviknal I am going to pick your commit, resolve conflicts and send another PR to merge this commit with the potential changes to rebase it.

**Comment by lucasponce** [comment]
Closed in favor of https://github.com/kiali/kiali-ui/pull/587 all comments and suggestions for this PR are addressed in the new one.

**Pull request closed without merging**